client/foundries_el2g.go

```diff
@@ -0,0 +1,47 @@
+package client
+
+import (
+	"encoding/json"
+)
+
+type El2gCsr struct {
+	Id    int    `json:"id"`
+	Value string `json:"value"`
+}
+
+func (a *Api) El2gCreateDg(factory string) (El2gCsr, error) {
+	url := a.serverUrl + "/ota/factories/" + factory + "/el2g/device-gateway/"
+	body, err := a.Post(url, []byte(""))
+	if err != nil {
+		return El2gCsr{}, err
+	}
+	var csr El2gCsr
+	err = json.Unmarshal(*body, &csr)
+	return csr, err
+}
+
+func (a *Api) El2gUploadDgCert(factory string, caId int, rootCa, cert string) error {
+	url := a.serverUrl + "/ota/factories/" + factory + "/el2g/device-gateway/"
+	type Cert struct {
+		Id                int    `json:"id"`
+		RootCa            string `json:"root-ca"`
+		SignedCertificate string `json:"signed-cert"`
+	}
+
+	body, err := json.Marshal(Cert{
+		Id:                caId,
+		RootCa:            rootCa,
+		SignedCertificate: cert,
+	})
+	if err != nil {
+		return err
+	}
+	_, err = a.Put(url, body)
+	return err
+}
+
+func (a *Api) El2gDeleteDg(factory string) error {
```

Comment:
Is it use anywhere?

Reply:
good catch b7d34016ff64f205b967e242b2f8d6c4824526f0